src/value.rs

```diff
@@ -283,7 +293,15 @@ pub mod testing {
 
     prop_compose! {
         /// Generate an arbitrary value in the range of valid nonnegative Zcash amounts.
-        pub fn arb_note_value()(value in 0u64..(MAX_MONEY as u64)) -> NoteValue {
+        pub fn arb_note_value()(value in 0u64..MAX_NOTE_VALUE) -> NoteValue {
+            NoteValue(value)
+        }
+    }
+
+    prop_compose! {
+        /// Generate an arbitrary value in the range of valid positive Zcash amounts
+        /// less than a specified value.
+        pub fn arb_nonnegative_note_value(max: u64)(value in 0u64..max) -> NoteValue {
```

Comment:
Note values are by definition non-negative. Let's name this `arb_bounded_note_value`.